# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: gfx/wr/wrench/reftests/gradient/linear-far-endpoints.yaml
# Commit: 6fc108de5c8f
# Full Hash: 6fc108de5c8fd527d0f5b6ae7b91604cc22e7d62
# Author: Nicolas Silva <nsilva@mozilla.com>
# Date: 2021-05-03 15:32:34
# Description:
#   Bug 1707744 - Avoid far gradient endpoints causing large gradient segments. r=gfx-reviewers,lsalzman
#   
#   Large segment bounds trip an assertion when casting coordinates to integers. Clipping early also reduces the amount of cached pixels.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113819
# ==============================================================================

diff -r 5d436e803f68 -r 6fc108de5c8f gfx/wr/wrench/reftests/gradient/linear-far-endpoints.yaml
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/gfx/wr/wrench/reftests/gradient/linear-far-endpoints.yaml	Mon May 03 09:12:42 2021 +0000
@@ -0,0 +1,10 @@
+# Axis-aligned linear gradient with very far endpoints. It goes through the gradient
+# decomposition path which should not choke on overflow or casting failure.
+---
+root:
+  items:
+    - type: gradient
+      bounds: 50 50 500 500
+      start: -19958788096 0
+      end: 19958788096 0
+      stops: [0.0, red, 1.0, blue]